operator/controllers/kyma_controller.go

```diff
@@ -206,6 +210,16 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alph
 			fmt.Errorf("error while syncing conditions during processing: %w", err))
 	}
 
+	status.Helper(r).SyncModuleInfo(kyma, modules)
```

Comment:
Why do you need a dedicated update on remove if you already have a sync here? IMO the next few lines are hard to understand in terms of naming

Reply:
If you check the SyncModuleInfo definition, it's just sync all module defined in `spec.modules`, and this sync doesn't require to call apiserver, that's why it happens both in `HandleConsistencyChanges` `HandleProcessingState`. 

And next few lines 
`UpdateStatusModuleInfos`: check if module resource still exists in cluster, if not exists, return true so that triggering `UpdateStatusForExistingModules` to update `status.moduleInfos`
`DeleteNoLongerExistingModules`: delete module which not decared in `spec.modules` but in `status.moduleInfos`
The above functions both require to call apiserver, only happens in `HandleProcessingState`

Reply:
With Sync we actually mean copy over then right? Maybe we change that name then to make it clearer?

For UpdateStatusModuleInfos I would maybe name it to UpdateModuleInfoStatus
for DeleteNoLongerExistingMoudles I think its okay :)